apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -0,0 +1,171 @@
+
+    if (!!paramModule && modulenames.includes(paramModule)) {
+        return paramModule;
+    } else {
+        return modulenames[0];
+    }
+};
+
+function ModuleViewWrapper({
+    id,
+    selectedModuleName,
+    modules,
+}: ModuleViewWrapperProps) {
+    const selectedModuleData = modules.find(
+        ([name]) => name === selectedModuleName
+    );
+
+    if (selectedModuleData) {
+        const [name, code] = selectedModuleData;
+
+        return <ModuleView id={id} name={name} code={code} />;
+    }
+
+    return <div />;
+}
+
+function PkgModuleViewWrapper({ id, modules }: Props) {
+    const modulenames = modules.map(([name], idx) => name);
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [query, setQuery] = useState('');
+    const [selectedModule, setSelectedModule] = useState(
+        initialSelectModule(searchParams, modulenames)
+    );
+
+    const filteredModules =
+        query === ''
+            ? modulenames
+            : modules
+                  .filter(([name]) => name.startsWith(query))
+                  .map(([name]) => name);
+
+    useEffect(() => {
+        setSearchParams({ module: selectedModule });
+    }, [selectedModule, setSearchParams]);
+
+    const submitSearch = useCallback(() => {
+        setSelectedModule((prev: string) =>
+            filteredModules.length === 1 ? filteredModules[0] : prev
+        );
+    }, [filteredModules]);
+
+    const clickModuleName = useCallback(
+        (moduleName: string) => () => {
+            setSelectedModule(moduleName);
+        },
+        []
+    );
+
+    return (
+        <div className="flex flex-col md:flex-row md:flex-nowrap gap-5 border-0 border-y border-solid border-sui-grey-45">
+            <div className="w-full md:w-1/5">
+                <Combobox value={selectedModule} onChange={setSelectedModule}>
+                    <div className="box-border border border-sui-grey-50 border-solid rounded-md shadow-sm placeholder-sui-grey-65 pl-3 w-full flex mt-2.5 justify-between py-2">
+                        <Combobox.Input
+                            onChange={(event) => setQuery(event.target.value)}
+                            displayValue={() => query}
+                            placeholder="Search"
+                            className="border-none w-full"
+                        />
+                        <button
+                            onClick={submitSearch}
+                            className="bg-inherit border-none pr-2"
+                            type="submit"
+                        >
+                            <SearchIcon className="fill-sui-steel cursor-pointer" />
+                        </button>
+                    </div>
+                    <Combobox.Options
+                        as="div"
+                        className="overflow-auto absolute left-0 w-full box-border md:w-1/6 md:left-auto h-fit shadow-moduleOption rounded-md bg-white z-10 px-2 pb-5 pt-3"
```

Comment:
I feel like this should have some sort of max height and scroll. When there's lots of modules this UI gets very big and unwieldy.